pilot/pkg/bootstrap/server.go

```diff
@@ -314,6 +316,24 @@ func NewServer(args *PilotArgs) (*Server, error) {
 
 	caOpts.Authenticators = authenticators
 	if features.XDSAuth {
+		authenticators := []security.Authenticator{
```

Comment:
Shouldn't the CA authenticator also add the OIDC auth ? I don't think there is any use case where we want a different set.

Reply:
The OIDC auth is added as the CA authenticator in pilot/pkg/bootstrap/istio_ca.go:
		jwtRule := v1beta1.JWTRule{Issuer: iss, Audiences: []string{aud}}
		oidcAuth, err := authenticate.NewJwtAuthenticator(&jwtRule, opts.TrustDomain)

Reply:
So why do we need duplicated code ? I meant: we should have a single code that initializes the list, and use it in both places, else they'll get out of sync and is harder to maintain.

Reply:
This PR does not change the existing CA auth flow. We can address the duplicate initialization in a separate PR.

Reply:
The authenticators should be exactly the same. please do not make them different.

Reply:
Regarding to the review comment "The authenticators should be exactly the same.", the JWT_RULE is for XDS authentication, not for CA. Therefore, the authenticators for XDS and CA are not exactly the same.

Reply:
The PR has been revised based on the review comments such that XDS and CA use the same authenticators.